src/Interpreters/SessionLog.cpp

```diff
@@ -132,8 +133,8 @@ NamesAndTypesList SessionLogElement::getNamesAndTypes()
         {"event_time", std::make_shared<DataTypeDateTime>()},
         {"event_time_microseconds", std::make_shared<DataTypeDateTime64>(6)},
 
-        {"user", std::make_shared<DataTypeString>()},
-        {"auth_type", std::move(identified_with_column)},
+        {"user", std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
```

Comment:
We can use empty string instead of NULL, don't we? Because we're going to prohibit empty user name anyway.

Reply:
I don't have a strong opinion about that, NULL might look ok too